drivers/charger/charger_pf1550.c

```diff
@@ -0,0 +1,695 @@
+	} else {
+		if (data->charger_status_notifier != NULL) {
+			data->charger_status_notifier(data->charger_status);
+		}
+	}
+
+	ret = pf1550_get_charger_online(data->dev, &data->charger_online);
+	if (ret < 0) {
+		LOG_WRN("Failed to read charger online %d", ret);
+	} else {
+		if (data->charger_online_notifier != NULL) {
+			data->charger_online_notifier(data->charger_online);
+		}
+	}
+
+	if (data->charger_online != CHARGER_ONLINE_OFFLINE) {
+		(void) pf1550_update_properties(data->dev);
+	}
+
+	ret = k_work_reschedule(&data->int_enable_work, INT_ENABLE_DELAY);
+	if (ret < 0) {
+		LOG_WRN("Could not reschedule int_enable_work: %d", ret);
+	}
+}
+
+static void pf1550_int_enable_work_handler(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct charger_pf1550_data *data = CONTAINER_OF(dwork, struct charger_pf1550_data,
+							  int_enable_work);
+
+	(void) pf1550_enable_interrupt_pin(data->dev, true);
+}
+
+static int pf1550_configure_interrupt_pin(const struct device *dev)
+{
+	struct charger_pf1550_data *data = dev->data;
+	const struct charger_pf1550_config *config = dev->config;
+	int ret;
+
+	if (!gpio_is_ready_dt(&config->int_gpio)) {
+		LOG_ERR("Interrupt GPIO device not ready");
+		return -ENODEV;
+	}
+
+	ret = gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
+	if (ret < 0) {
+		LOG_ERR("Could not configure interrupt GPIO");
+		return ret;
+	}
+
+	gpio_init_callback(&data->gpio_cb, pf1550_gpio_callback, BIT(config->int_gpio.pin));
+	ret = gpio_add_callback_dt(&config->int_gpio, &data->gpio_cb);
+	if (ret < 0) {
+		LOG_ERR("Could not add interrupt GPIO callback");
+		return ret;
+	}
+
+	return 0;
+}
+
+static int pf1550_init(const struct device *dev)
+{
+	struct charger_pf1550_data *data = dev->data;
+	const struct charger_pf1550_config *config = dev->config;
+	int ret;
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+
+	ret = pf1550_init_properties(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	k_work_init(&data->int_routine_work, pf1550_int_routine_work_handler);
+	k_work_init_delayable(&data->int_enable_work, pf1550_int_enable_work_handler);
+
+	ret = pf1550_configure_interrupt_pin(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = pf1550_enable_interrupt_pin(dev, true);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = pf1550_enable_interrupts(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to enable interrupts");
+		return ret;
+	}
+
+	ret = pf1550_update_properties(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to setup charger");
+		return ret;
+	}
+
+	return 0;
+}
+
+static const struct charger_driver_api pf1550_driver_api = {
+	.get_property = pf1550_get_prop,
+	.set_property = pf1550_set_prop,
+	.charge_enable = pf1550_set_enabled,
+};
+
+/*
+#define PF1550_LED_DEFINE(inst)								\
+	static struct charger_pf1550_led_config_##inst = {				\
+		.enabled = DT_INST_PROP(inst, constant_charge_current_max_microamp),	\
+	}
+
+	TODO: add led_config field to dt
```

Comment:
yeah I think getting the LED configuration from the devicetree is fine, just put some enums with the possible options